File: gridTraveler memoization/gridTraveler_memoization/gridTraveler_memoization/gridTraveler_memoization.cpp
#include <iostream>
#include <unordered_map>

using namespace std;

unordered_map<long long, long long> memo;

long long grid_traveler(int m, int n) {
    long long key = m * 1000000LL + n;
    if (memo.find(key) != memo.end()) {
        return memo[key];
    }
    if (m == 1 && n == 1) {
        return 1;
    }
    if (m == 0 || n == 0) {
        return 0;
    }
    memo[key] = grid_traveler(m - 1, n) + grid_traveler(m, n - 1);
    return memo[key];
}

int main() {
    cout << grid_traveler(18, 18) << std::endl;
    return 0;
}

File: gridTraveler memoization/gridTravelermemoization.py
def gridTraveler_memoization(m,n,memo={}):
    key = f"{m},{n}"
    if key in memo :
        return memo[key]
    if m==1 and n==1:
        return 1
    if m == 0 or n == 0 :
        return 0
    memo[key]  = gridTraveler_memoization(m-1 , n ,memo) + gridTraveler_memoization(m , n-1,memo)
    return memo[key]

print(gridTraveler_memoization(18,18))